x-pack/legacy/plugins/alerting/server/alerts_client.test.ts

```diff
@@ -12,24 +12,33 @@ import { alertTypeRegistryMock } from './alert_type_registry.mock';
 import { TaskStatus } from '../../task_manager';
 import { IntervalSchedule } from './types';
 import { resolvable } from './test_utils';
+import { encryptedSavedObjectsMock } from '../../../../plugins/encrypted_saved_objects/server/mocks';
 
 const taskManager = taskManagerMock.create();
 const alertTypeRegistry = alertTypeRegistryMock.create();
 const savedObjectsClient = savedObjectsClientMock.create();
+const encryptedSavedObjects = encryptedSavedObjectsMock.createStart();
 
 const alertsClientParams = {
   taskManager,
   alertTypeRegistry,
   savedObjectsClient,
   spaceId: 'default',
+  namespace: 'default',
   getUserName: jest.fn(),
   createAPIKey: jest.fn(),
+  invalidateAPIKey: jest.fn(),
   logger: loggingServiceMock.create().get(),
+  encryptedSavedObjectsPlugin: encryptedSavedObjects,
 };
 
 beforeEach(() => {
   jest.resetAllMocks();
-  alertsClientParams.createAPIKey.mockResolvedValue({ created: false });
+  alertsClientParams.createAPIKey.mockResolvedValue({ apiKeysEnabled: false });
+  alertsClientParams.invalidateAPIKey.mockResolvedValue({
+    apiKeysEnabled: true,
+    result: { error_count: 0 },
+  });
```

Comment:
Feels like we're testing the happy path here, it might be worth adding a couple of tests that checks what happens when these api calls fail (throw an error, etc)